regression_tests/regression_workflow_test.go

```diff
@@ -0,0 +1,332 @@
+//go:build regression
+// +build regression
+
+package regressiontests
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/commands"
+)
+
+func TestRegressionWorkflow(t *testing.T) {
+	ctx := context.Background()
+	spvWalletPG, spvWalletSL := initSPVWalletAPIs(t)
+
+	t.Log("Step 1: Setup success: created SPV client instances with test users")
+	t.Logf("SPV clients for env: %s, user: %s, admin: %s, leader: %s", spvWalletPG.cfg.envURL, spvWalletPG.user.alias, spvWalletPG.admin.alias, spvWalletPG.leader.alias)
+	t.Logf("SPV clients for env: %s, user: %s, admin: %s, leader: %s", spvWalletSL.cfg.envURL, spvWalletSL.user.alias, spvWalletSL.admin.alias, spvWalletSL.leader.alias)
+
+	t.Run("Step 2: The leader user attempts to retrieve the shared configuration response from their SPV Wallet API instance.", func(t *testing.T) {
+		tests := []struct {
+			name                string
+			leader              *user
+			user                *user
+			admin               *admin
+			expectedPaymailsLen int
+		}{
+			{
+				name:   fmt.Sprintf("%s should set paymail domain after fetching shared config", spvWalletPG.leader.alias),
+				leader: spvWalletPG.leader,
+				user:   spvWalletPG.user,
+				admin:  spvWalletPG.admin,
+			},
+			{
+				name:   fmt.Sprintf("%s should set paymail domain after fetching shared config", spvWalletSL.leader.alias),
+				leader: spvWalletSL.leader,
+				user:   spvWalletSL.user,
+				admin:  spvWalletSL.admin,
```

Comment:
Why not simply:
`server: spvWalletSL`

It would make it simple (and when I saw it at the begining of the test, I was wandering why you need all those users to get the paymail domain by leader)